p4proto/p4src/linux_networking/README.md

```diff
@@ -0,0 +1,212 @@
+```
+#### 6. Spawn two VM's on vhost-user ports created in step 3, start the VM's and assign IP's
+```
+    ip addr add 99.0.0.1/24 dev eth0
+    ip link set dev eth0 up
+    ip addr add 99.0.0.2/24 dev eth0
+    ip link set dev eth0 up
+```
+#### 7. Bring up the TAP interfaces and assign IP to TAP interfaces
+```
+    ip link set dev TAP0 up
+    ip addr add 40.1.1.1/24 dev TAP1
+    ip link set dev TAP1 up
+    ip link set dev TAP2 up
+    ip link set dev TAP3 up
+```
+
+#### 8. Set the pipeline and add br-int to OVS
+```
+    ovs-p4ctl set-pipe br0 lnw.pb.bin p4Info.txt
+    ovs-vsctl add-br br-int
+    ip link set dev br-int up
+```
+
+#### 9. Configure VXLAN port
+```
+    ovs-vsctl add-port br-int vxlan1 -- set interface vxlan1 type=vxlan options:local_ip=40.1.1.1 options:remote_ip=40.1.1.2 options:dst_port=4789
+Note:
+    - VXLAN destination port should always be standard port. i.e., 4789. (limitation by p4 parser)
+```
+#### 10. Configure VLAN ports on TAP0 and add them to br-int
+```
+    ip link add link TAP0 name vlan1 type vlan id 1
+    ip link add link TAP0 name vlan2 type vlan id 2
+    ovs-vsctl add-port br-int vlan1
+    ovs-vsctl add-port br-int vlan2
+    ip link set dev vlan1 up
+    ip link set dev vlan2 up
+
+Note: 
+     - All VLAN interfaces should be created on top of TAP ports, and should always be in lowercase format "vlan+vlan_id. (Eg: vlan1, vlan2, vlan3 …. vlan4094)
+```
+
+#### 11. Configure rules to push and pop VLAN from vhost 0 and 1 ports to TAP0 port (vhost-user and vlan port mapping)
+```
+Note: Port number used in ovs-p4ctl commands are target datapath indexes(unique identifier for each port) which can be queried using following commands below. With current SDE implementation, both tdi-portin-id and tdi-portout-id are the same.
+
+    gnmi-cli get "device:virtual-device,name:net_vhost0,tdi-portin-id"
+    gnmi-cli get "device:virtual-device,name:net_vhost0,tdi-portout-id"
+
+    Target DP index of control TAP port will be Target DP index of corresponding physical port + 1. If the ports are created in the order mentioned in the above step, target datapath indexes will be:
+
+                   Port name          Target datapath index
+                   vhost-user-0(VM1) - 0
+                   vhost-user-1(VM2) - 1
+                   phy-port0         - 2
+                   TAP1              - 3
+                   phy-port1         - 4
+                   TAP2              - 5
+                   TAP0              - 6
+                   TAP3              - 7
+```
+ - **Rule for: Any tx control packet from VM1(TDP 0), pipeline should add a VLAN tag 1 and send it to TAP0 port(TDP 6)**
+  ```
+  ovs-p4ctl add-entry br0 linux_networking_control  .handle_tx_control_pkts_table "istd.input_port=0,action=linux_networking_control.push_vlan_fwd(6,1)"
```

Comment:
please remove space between for the table name
linux_networking_control & handle_tx_control_pkts_table